File: Factory/Java/DatabaseAdapter/DatabaseFactory/src/com/laxmena/db/MongoDatabase.java
package com.laxmena.db;

public class MongoDatabase implements Database {
    public MongoDatabase() {
        System.out.println("============= MongoDB =============");
    }
    @Override
    public void createTable(String tableName) {
        System.out.println("Creating table now!! " + tableName);
    }

    @Override
    public void queryTable(String tableName, String query) {
        System.out.println("Table: " + tableName + " | Query: " + query);
    }
}

File: Factory/Python/DatabaseAdapter/DatabaseAdapter/db/mongo_database.py
from db.database import Database


class MongoDatabase(Database):
    def __init__(self):
        print("======== MongoDB ========")

    def create_table(self, table_name: str) -> None:
        print("Table Creation ==> " + table_name)

    def query_table(self, table_name: str, fields: str) -> None:
        print("Querying Table: " + table_name + " for fields " + fields)
